tests/components/template/test_config_flow.py

```diff
@@ -804,7 +804,7 @@ async def test_config_flow_preview(
                 ),
                 "unit_of_measurement": (
                     "'None' is not a valid unit for device class 'energy'; "
-                    "expected one of 'cal', 'Gcal', 'GJ', 'GWh', 'J', 'kcal', 'kJ', 'kWh', 'Mcal', 'MJ', 'MWh', 'TWh', 'Wh'"
+                    "expected one of 'cal', 'Gcal', 'GJ', 'GWh', 'J', 'kcal', 'kJ', 'kWh', 'Mcal', 'MJ', 'mWh', 'MWh', 'TWh', 'Wh'"
```

Comment:
Hm, this test did pass locally.

Now running it multiple times, it seems the order is random here.

The reason is that we sort the units with `str.casefold`:

https://github.com/home-assistant/core/blob/4e5df1d854a4926bedc474aa9c16ff70057ebbdc/homeassistant/components/template/config_flow.py#L240

Which made me realize that the new unit `mWh` is actually equal to `MWh` when used case insensitivity. Is that acceptable?

From what I understand this is really the [SI/metric prefixes](https://en.wikipedia.org/wiki/Metric_prefix). I guess we might run into this in other places too, but currently I don't see any precedence :thinking: 

I try changing the above sort and see how things are going.